code/controllers/ErrorPageControllerExtension.php

```diff
@@ -9,7 +9,8 @@
 class ErrorPageControllerExtension extends Extension {
 
 	public function onBeforeHTTPError($statusCode, $request) {
-		$response = ErrorPage::response_for($statusCode);
+		// Don't use the HTML response for media requests
+		$response = $request->isMedia() ? null : ErrorPage::response_for($statusCode);
```

Comment:
This is merged from the ContentController::httpResponse() method. I assume it's to prevent page rendering overhead, like if you have a page with a bunch of broken links to static resources and you're not caching your 404 response effectively it could cause some stress on the server because each page load would result in 40 error page rendering cycles, which are redundant because the user will not see them.

I assumed I was following the current functionality here but actually I've just tested and static resources are receiving a full error page response prior to this change. Following this change you would get only a plain text error message for static resources. It looks like we have some intended functionality here that isn't currently working - my guess is that checking whether a request is media or not in ContentController::httpError() is too late in the chain as an error would be thrown before we get to this point? In that case I'm not sure if it's best to fix this once-intended behaviour, or remove it.